soroban-sdk/src/tests/budget.rs

```diff
@@ -24,6 +24,9 @@ fn test_budget() {
     let b = client.add();
     e.budget().print();
 
-    assert_eq!(e.budget().tracker(ContractCostType::VisitObject), (4, None));
+    assert_eq!(
+        e.budget().tracker(ContractCostType::VisitObject),
+        (13, None)
+    );
```

Comment:
Surprising large jump in visit object in this upgrade. cc @jayz22 @graydon

Reply:
Yeah, I added a bunch of integrity checking when passing and returning arguments: https://github.com/stellar/rs-soroban-env/pull/1221